Bunch of threads throwing IllegalStateException

I was running Bulk Rwalk test on a Uno cluster with 2 tservers. I killed one of the tservers (with kill -9) and then restarted it (with `accumulo-cluster start-tservers`). A little while later 50+ of these errors flooded the Monitor. 
<pre>
java.lang.RuntimeException: java.lang.IllegalStateException: Unexpected family chopped
	at org.apache.accumulo.tserver.AssignmentHandler.run(AssignmentHandler.java:141)
	at org.apache.accumulo.tserver.ActiveAssignmentRunnable.run(ActiveAssignmentRunnable.java:63)
	at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalStateException: Unexpected family chopped
	at org.apache.accumulo.core.metadata.schema.TabletMetadata.convertRow(TabletMetadata.java:403)
	at org.apache.accumulo.core.metadata.schema.TabletsMetadata$Builder.lambda$buildNonRoot$8(TabletsMetadata.java:216)
	at com.google.common.collect.Iterators$6.transform(Iterators.java:829)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:52)
	at com.google.common.collect.Iterators$5.computeNext(Iterators.java:672)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.apache.accumulo.core.metadata.schema.AmpleImpl.readTablet(AmpleImpl.java:47)
	at org.apache.accumulo.core.metadata.schema.Ample.readTablet(Ample.java:144)
	at org.apache.accumulo.tserver.AssignmentHandler.run(AssignmentHandler.java:109)
	... 6 more
</pre>

I think these errors are related to a Chop compaction that was taking place when the tserver was killed.
<pre>
2022-09-26T10:36:12,167 [manager.Manager] INFO : Asking localhost:10000[100007556480005] to chop a;r0265c;r01179
tserver2_ip-10-113-14-231.log:2022-09-26T10:36:24,770 [tablet.files] DEBUG: Compacting a;r0265c;r01179 on i.default.small for CHOP from [I00002vg.rf, I00002it.rf, I00002h5.rf, C00001an.rf, I00002xr.rf, I0000303.rf, I00002zo.rf, I00002vh.rf, I00002hd.rf, I00002ob.rf] size 36 KB
tserver2_ip-10-113-14-231.log:2022-09-26T10:36:31,369 [tablet.files] DEBUG: Compacted a;r0265c;r01179 for CHOP created hdfs://localhost:8020/accumulo/tables/a/t-00002pq/C000033u.rf from [I00002oe.rf, I00002nv.rf, I00002nw.rf, I00002no.rf, I00002oh.rf, I00002nb.rf, I00002nj.rf, I00002oc.rf, I00002m2.rf]
...
2022-09-26T10:36:36,121 [manager.Manager] ERROR: unable to get tablet server status localhost:10000[100007556480005]
2022-09-26T10:37:03,458 [manager.Manager] DEBUG: 85 assigned to dead servers: ...  a;r0265c;r01179
2022-09-26T10:37:03,468 [tablet.location] DEBUG: Suspended a;r0265c;r01179 to localhost:10000 at 3696264 ms with 1 walogs
...
2022-09-26T10:37:37,537 [manager.Manager] INFO : New servers: [localhost:10000[10000755648001d]]
2022-09-26T10:37:37,539 [manager.EventCoordinator] INFO : There are now 2 tablet servers
2022-09-26T10:37:37,683 [tablet.location] DEBUG: Assigned a;r0265c;r01179 to localhost:10000[10000755648001d]
2022-09-26T10:37:41,266 [manager.Manager] ERROR: localhost:10000 reports assignment failed for tablet a;r0265c;r01179
</pre>

--- a/core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java
+++ b/core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java
@@ -326,8 +326,9 @@ public class MetadataSchema {
      * data for the current tablet, so that they are safe to merge
      */
     public static class ChoppedColumnFamily {
-      public static final Text NAME = new Text("chopped");
-      public static final ColumnFQ CHOPPED_COLUMN = new ColumnFQ(NAME, new Text("chopped"));
+      public static final String STR_NAME = "chopped";
+      public static final Text NAME = new Text(STR_NAME);
+      public static final ColumnFQ CHOPPED_COLUMN = new ColumnFQ(NAME, new Text(STR_NAME));
     }
 
     public static class ExternalCompactionColumnFamily {

--- a/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java
+++ b/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java
@@ -54,6 +54,7 @@ import org.apache.accumulo.core.metadata.TabletFile;
 import org.apache.accumulo.core.metadata.TabletLocationState;
 import org.apache.accumulo.core.metadata.TabletState;
 import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.BulkFileColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.ChoppedColumnFamily;
 import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.ClonedColumnFamily;
 import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.CurrentLocationColumnFamily;
 import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.DataFileColumnFamily;
@@ -106,6 +107,7 @@ public class TabletMetadata {
   private OptionalLong compact = OptionalLong.empty();
   private Double splitRatio = null;
   private Map<ExternalCompactionId,ExternalCompactionMetadata> extCompactions;
+  public boolean chopped = false;
 
   public enum LocationType {
     CURRENT, FUTURE, LAST
@@ -398,6 +400,9 @@ public class TabletMetadata {
         case ExternalCompactionColumnFamily.STR_NAME:
           extCompBuilder.put(ExternalCompactionId.of(qual),
               ExternalCompactionMetadata.fromJson(val));
+          break;
+        case ChoppedColumnFamily.STR_NAME:
+          te.chopped = true;
           break;
         default:
           throw new IllegalStateException("Unexpected family " + fam);

--- a/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java
+++ b/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java
@@ -107,7 +107,7 @@ public class TabletMetadata {
   private OptionalLong compact = OptionalLong.empty();
   private Double splitRatio = null;
   private Map<ExternalCompactionId,ExternalCompactionMetadata> extCompactions;
-  public boolean chopped = false;
+  private boolean chopped = false;
 
   public enum LocationType {
     CURRENT, FUTURE, LAST
@@ -129,6 +129,7 @@ public class TabletMetadata {
     COMPACT_ID,
     SPLIT_RATIO,
     SUSPEND,
+    CHOPPED,
     ECOMP
   }
 
@@ -265,6 +266,11 @@ public class TabletMetadata {
   public Double getSplitRatio() {
     ensureFetched(ColumnType.SPLIT_RATIO);
     return splitRatio;
+  }
+
+  public boolean hasChopped() {
+    ensureFetched(ColumnType.CHOPPED);
+    return chopped;
   }
 
   public SortedMap<Key,Value> getKeyValues() {

